firmware/thruna/PDM/src/app/app_retryHandler.h

```diff
@@ -0,0 +1,43 @@
+#include <stdbool.h>
+#pragma once
+
+typedef enum
+{
+    RETRY_PROTOCOL_LV,
+    RETRY_PROTOCOL_DI_LHS,
+    RETRY_PROTOCOL_DI_RHS,
+    RETRY_PROTOCOL_AIR,
+    RETRY_PROTOCOL_AUX,
+    RETRY_PROTOCOL_FANS,
+    RETRY_PROTOCOL_EMETER,
+    RETRY_PROTOCOL_DRS,
+    NUM_RETRY_PROTOCOL
+} RetryProtocol;
+
+typedef enum
+{
+    RETRY_STATE_OFF,
+    RETRY_STATE_RUNNING,
+    RETRY_STATE_EXPIRED,
+    RETRY_STATE_WAITING
+} RetryState;
+
+typedef struct
+{
+    bool          efuse_state;
+    RetryProtocol retry_protocol;
+    int           retry_attempts_limit;
+} RetryConfig;
+
+typedef struct
+{
+    RetryState retry_state;
+    int        retry_attempts;
+    int        timer_attempts;
+    int        timer_attempts_limit;
+    float      current_sum;
+} RetryData;
+
+void retry_handler_start(RetryProtocol protocol, const RetryConfig retry_configs[], RetryData retry_data[]);
+
+void retry_handler_recover(RetryProtocol protocol, const RetryConfig retry_configs[], RetryData retry_data[]);
```

Comment:
Not sure I understand how these 2 functions work. It seems like when we've decided an efuse is faulted, we call `retry_handler_start`, which looks like in a lot of cases it just disables the efuses.

Then there is `retry_handler_recover`, which we call if we decide the efuses aren't faulted, which is the standby reset we've talked about. However, I would think that if we decide the efuses aren't faulted, we shouldn't do anything?

I think I'm probably missing something, maybe we should talk about this on Monday.

Reply:
Also nitpick: `app_retryHandler_start` and `app_retryHandler_recover`